x/gov/client/proposal_handler.go

```diff
@@ -0,0 +1,26 @@
+package client
+
+import (
+	"github.com/spf13/cobra"
+
+	"github.com/cosmos/cosmos-sdk/client/context"
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/x/gov/client/rest"
+)
+
+// function to create the rest handler
+type RESTHandlerFn func(context.CLIContext, *codec.Codec) rest.ProposalRESTHandler
```

Comment:
Errr shall we not just define this `ProposalRESTHandler` type in the `rest` package? Has nothing to do specifically with gov, right?

Reply:
I'm confused `ProposalRESTHandler` _is_ already defined in `	"github.com/cosmos/cosmos-sdk/x/gov/client/rest"`  - if you were referring to the `RESTHandlerFn` this _should_ stay where it is, it's only used for defining the `ProposalHandler` (which belongs outside of rest because its used for both the cli and rest)